doc/releases/changelog-dev.md

```diff
@@ -408,6 +408,9 @@
 
 <h3>Internal changes ⚙️</h3>
 
+* Now `TransformProgram` can dispatch to anything individual transforms can dispatch onto.
+  [(#8731)](https://github.com/PennyLaneAI/pennylane/pull/8731)
+
```

Comment:
Actually, I would also recommend putting merging this into the same block as the other transform docstrings. This is *most definitely* not an internal change, but is actually a headline feature.